doc/Discriminator-support.md

```diff
@@ -0,0 +1,91 @@
+# Discriminator support
+
+`@exodus/schemasafe` supports a strict subset of the [`discriminator` API](FIXME).
+
+It has the following additional requirements to be safe and compatible with upstream JSON Schema
+specification:
+
+  1. At least one of `oneOf` or `anyOf` is required. Using both at the same time is not allowed.
+
+  2. Type of the object using `discriminator` must be provable to be `object` — can be specified
+     either as `type` on the same level as `discriminator`, or in each branch separately.
+
+  3. Each `oneOf`/`anyOf` branch **must** have a `const` rule on the property targeted by
+    `propertyName` of the `discriminator`, either directly or inside a `$ref`.
+
+    _While seeming a bit excessive, this is the rule that makes `discriminator` both well-defined
+    and compatible with JSON Schema spec, while being a subset of OAPI `discriminator`._
+
+    Currently, those `const` values must be unique strings, which additionally makes `oneOf` and
+    `anyOf` identical there in relation to the validation result. The uniqueness and being a string
+    requirement might be lifted in the future, if it would be deemed useful.
+
+  4. Property targeted by `propertyName` of the `discriminator` **must** be placed in `required`,
+     either on the same level as `discriminator`, or in each branch separately. Failing to do so
+     would make _all_ `const` checks pass on that property, per the JSON Schema specification.
+
+  5. `mapping` is supported but only when it has the exact same set of branches as `oneOf`/`anyOf`,
+    values of the `mapping` correspond to used `$ref` values of the branches and the keys of
+    the mapping match `const` values on the `propertyName` of the `discriminator` in corresponding
+    branches.
+
+    That way, `mapping` doesn't really do anything at all, and brings in no new information to the
+    validator. It only serves a sanity check (`@exodus/schemasafe` will refuse to compile a schema
+    which has a `mapping` that mismatches the above constraints) and as an additional commentary
```

Comment:
```suggestion
    which has a `mapping` that mismatches the above constraints) and has an additional commentary
```

Typo?

Reply:
Probably not here, but `It only serves a sanity check` -> `It only serves as a sanity check .... and as an`.
Not sure how to reword that.